src/tdastro/utils/bicubic_interp.py

```diff
@@ -0,0 +1,242 @@
+"""The BicubicInterpolator is used by SALT models.
+
+It is adapted from sncosmo's BicubicInterpolator class (but implemented in JAX):
+https://github.com/sncosmo/sncosmo/blob/v2.10.1/sncosmo/salt2utils.pyx
+"""
+
+import jax.numpy as jnp
+from jax import jit, vmap
+from jax.lax import cond
+
+from tdastro.utils.io_utils import read_grid_data
+
+
+@jit
```

Comment:
I'm not sure that we need `@jit` on this level, should we compile the code using cubic interpolation instead?